### ISSUE TITLE
Add hidden/private benchmark challenge workflow without exposing test data

### ISSUE THREAD

**Issue by dipeshbabu** [dipeshbabu/private-lab-bench#33]
## Why this matters

Public benchmark test sets can leak, be copied, or eventually appear in training data. Scientific benchmark owners may also be unable to release their real evaluation data because it is proprietary, unpublished, sensitive, expensive, or institutionally restricted.

PrivateLabBench should support **hidden scientific benchmarks** where the evaluator owns the test data and participants provide predictions or, later, executable model artifacts.

## Goal

Create a reproducible benchmark-owner workflow for evaluating submissions locally against a hidden test set and publishing only verifiable aggregate results.

## Initial scope

- [ ] Define a versioned hidden-benchmark specification that references an evaluation protocol but keeps dataset paths local-only.
- [ ] Support prediction-table submissions as the first interoperability boundary.
- [ ] Validate submission schema and stable sample IDs without revealing target values.
- [ ] Add a local `plb benchmark-evaluate` or equivalent owner command.
- [ ] Generate a shareable evaluation receipt for each submission.
- [ ] Bind model/submission identity, protocol version, benchmark ID/version, and artifact hash into the receipt.
- [ ] Add a local leaderboard/table builder that consumes only verified receipts.
- [ ] Support ranking rules only where the benchmark protocol explicitly defines them.
- [ ] Include uncertainty/tie handling once #27 is available rather than overinterpreting tiny metric differences.
- [ ] Add safeguards against accidental target/test-set disclosure in generated artifacts.
- [ ] Add optional submission manifest support for participant metadata/citation/model version.
- [ ] Document threat boundaries: this is hidden-data evaluation, not a complete anti-cheating or secure sandbox system.
- [ ] Design a future extension point for container/model execution without making arbitrary code execution part of the initial implementation.

## Target workflow

Participant:

```text
model → predictions.csv
```

Benchmark owner:

```bash
plb benchmark-evaluate hidden-admet.yaml predictions.csv
```

Published output:

```text
submission receipt + aggregate leaderboard entry
```

Never published:

```text
test rows / targets / private metadata / local paths
```

## Acceptance criteria

A research group can maintain a non-public scientific test set, evaluate external prediction submissions under a fixed protocol, and publish independently verifiable result artifacts without publishing the hidden evaluation data.
